src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/service/IncentiveMappingService.kt

```diff
@@ -27,23 +27,51 @@ class IncentiveMappingService(
     private val log = LoggerFactory.getLogger(this::class.java)
   }
 
+  fun alreadyExistsMessage(
+    duplicateMapping: IncentiveMappingDto,
+    existingMapping: IncentiveMappingDto,
+  ) =
+    "Incentive mapping already exists. \nExisting mapping: $existingMapping\nDuplicate mapping: $duplicateMapping"
+
   @Transactional
   suspend fun createIncentiveMapping(createMappingRequest: IncentiveMappingDto) =
     with(createMappingRequest) {
       log.debug("creating incentive $createMappingRequest")
       incentiveMappingRepository.findById(incentiveId)?.run {
-        if (this@run.nomisBookingId == this@with.nomisBookingId && this@run.nomisIncentiveSequence == this@with.nomisIncentiveSequence) {
-          log.debug("Incentive mapping already exists for nomisBookingId: $nomisBookingId and nomisIncentiveSequence: $nomisIncentiveSequence: $incentiveId so not creating. All OK")
```

Comment:
t some point I think we should make this whole checking code a bit neater - obviously nothing to do with this PR